management-account/terraform/cost-categories-business-unit.tf

```
@@ -85,38 +85,38 @@ resource "aws_ce_cost_category" "business_unit" {
   rule_version    = "CostCategoryExpression.v1"
   effective_start = "2025-02-01T00:00:00Z"
 
-  # Rule 1: Prioritize the Resource `business-unit` Cost Allocation Tag to assign cost
+  # Rule 0: Temporary Correction: Assign Cost Category Based on Account Prefixes
+  # Due to organisational structure change some accounts currently have the wrong `business-unit` tag
+  # and resource level BU tag for MP created accounts. This temporary rule allows us to correctly assign cost
+  # and give teams time to correct their account `business-unit` tags.
   dynamic "rule" {
-    for_each = { for k, v in local.business_units : k => v if length(v.business_unit_tag_values) > 0 }
+    for_each = { for k, v in local.business_units : k => v if length(v.untagged_aws_account_prefixes) > 0 }
     content {
       type  = "REGULAR"
       value = rule.key
 
       rule {
-        tags {
-          key           = "business-unit"
-          values        = rule.value.business_unit_tag_values
-          match_options = ["EQUALS"]
+        dimension {
+          key           = "LINKED_ACCOUNT_NAME"
+          values        = rule.value.untagged_aws_account_prefixes
+          match_options = ["STARTS_WITH"]
         }
       }
     }
   }
 
-  # Rule 1.5: Temporary Correction: Assign Cost Category Based on Account Prefixes
-  # Due to organisational structure change some accounts currently have the wrong `business-unit` tag
-  # so Rule 2 would assign cost incorrectly. This temporary rule allows us to correctly assign cost
-  # and give teams time to correct their account `business-unit` tags.
+  # Rule 1: Prioritize the Resource `business-unit` Cost Allocation Tag to assign cost
```

```suggestion
  # Rule 1: Use the Resource `business-unit` Cost Allocation Tag to assign cost
```
Can we change this again - as we are not prioritizing this rule anymore.

---

Looks good thanks